src/lib/cimbar_js/cimbar_recv_js.cpp

```diff
@@ -186,15 +244,29 @@ int cimbard_get_filename(const uchar* finbuffer, unsigned size, char* filename,
 	return fn.size();
 }
 
-// if fountain_decode returned a >0 value, call this to retrieve the reassembled file
-// bouth fountain_*() calls should be from the same js webworker/thread
-int cimbard_finish_copy(uint32_t id, uchar* buffer, unsigned size)
+int cimbard_decompress_read(uint32_t id, unsigned char* buffer, unsigned size)
 {
-	if (!_sink)
-		return -1;
-	if (!_sink->recover(id, buffer, size))
-		return -2;
-	return 0;
+	int	res = recover_contents(id);
```

Comment:
Notably both decompress_read() *and* get_filename() will call `recover_contents()` (result will be cached), since they both need to pull the data from the zstd stream. Definitionally, then, you can only call either after `cimbard_fountain_decode()` has finished reassembling the file.